doped/utils/parsing.py

```
@@ -1988,3 +2009,374 @@ def _multiple_files_warning(file_type, directory, chosen_filepath, action=None,
+
+
+
+from pymatgen.entries.computed_entries import ComputedEntry
+from doped.utils.parsing import parse_projected_eigen, find_archived_fname
+from pymatgen.io.espresso.outputs.pwxml import PWxml
+from ase.io.cube import read_cube_data
+from scipy.ndimage import map_coordinates
+BOHR_TO_ANGSTROM = 0.529177
+
+class RunParser:
+    def __new__(cls, code: Literal["vasp", "espresso"], **kwargs):
```

Ruff (0.12.2)

Missing docstring in public method

(D102)